packages/wrangler/src/generate.ts

```diff
@@ -1,33 +1,130 @@
 
+// https://github.com/cloudflare/wrangler/blob/master/src/cli/mod.rs#L106-L123
 interface GenerateArgs {
-	name: string;
-	template: string;
+	name?: string;
+	template?: string;
+	type?: string;
+	site?: boolean;
 }
 
 export function generateOptions(yargs: Argv) {
 	return yargs
 		.positional("name", {
 			describe: "Name of the Workers project",
-			default: "worker",
+			type: "string",
 		})
 		.positional("template", {
+			type: "string",
 			describe: "The URL of a GitHub template",
-			default: "https://github.com/cloudflare/worker-template",
+		})
+		.option("type", {
+			alias: "t",
+			type: "boolean",
+			hidden: true,
+			deprecated: true,
+		})
+		.option("site", {
+			alias: "s",
+			type: "boolean",
+			hidden: true,
+			deprecated: true,
 		});
 }
 
-export function generateHandler(
-	generateArgs: ArgumentsCamelCase<GenerateArgs>
-) {
-	// "👯 [DEPRECATED]. Scaffold a Cloudflare Workers project from a public GitHub repository.",
-	throw new DeprecationError(
-		"`wrangler generate` has been deprecated.\n" +
-			"Try running `wrangler init` to generate a basic Worker, or cloning the template repository instead:\n\n" +
+export async function generateHandler({
+	// somehow, `init` marks name as required but then also runs fine
+	// with the name omitted, and then substitutes it at runtime with ""
+	name = "",
+	template,
+	type,
+	site,
+	...args
+}: ArgumentsCamelCase<GenerateArgs>) {
+	// delegate to `wrangler init` if no template is specified
+	if (template === undefined) {
+		return initHandler({ name, ...args });
+	}
+
+	// print down here cuz `init` prints it own its own
+	printWranglerBanner();
+
+	if (type) {
+		let message = "The --type option is no longer supported.";
+		if (args.type === "webpack") {
+			message +=
+				"\nIf you wish to use webpack then you will need to create a custom build.";
+			// TODO: Add a link to docs
+		}
+		throw new CommandLineArgsError(message);
+	}
+
+	const creationDirectory = generatedWorkerDir(name);
+
+	if (site) {
+		const gitDirectory =
+			creationDirectory !== process.cwd()
+				? path.basename(creationDirectory)
+				: "my-site";
+		const message =
+			"The --site option is no longer supported.\n" +
+			"If you wish to create a brand new Worker Sites project then clone the `worker-sites-template` starter repository:\n\n" +
 			"```\n" +
-			`git clone ${generateArgs.template}\n` +
+			`git clone --depth=1 --branch=wrangler2 https://github.com/cloudflare/worker-sites-template ${gitDirectory}\n` +
+			`cd ${gitDirectory}\n` +
 			"```\n\n" +
-			"Please refer to https://developers.cloudflare.com/workers/wrangler/deprecations/#generate for more information."
+			"Find out more about how to create and maintain Sites projects at https://developers.cloudflare.com/workers/platform/sites.\n" +
+			"Have you considered using Cloudflare Pages instead? See https://pages.cloudflare.com/.";
+		throw new CommandLineArgsError(message);
+	}
```

Comment:
This is copied verbatim from `init`, maybe it's worth creating `SiteProjectCreationError` and `TypeArgDeprecationError`s?